src/inspect_ai/scorer/_metrics/std.py

```diff
@@ -60,12 +62,52 @@ def stderr(to_float: ValueToFloat = value_to_float()) -> Metric:
             float directly, and prints a warning and returns
             0 if the Value is a complex object (list or dict).
 
+        cluster (str | None): The key from the Sample metadata
+            corresponding to a cluster identifier, if clustered
+            standard errors are to be computed.
+
     Returns:
         stderr metric
     """
 
-    def metric(scores: list[Score]) -> float:
-        values = [to_float(score.value) for score in scores]
+    def clustered_metric(scores: list[SampleScore]) -> float:
+        """Computes a clustered standard error.
+
+        For details, see Appendix A of https://arxiv.org/pdf/2411.00640.
+        The version here uses a finite cluster correction (unlike the paper)
+        """
+        assert cluster is not None
+        cluster_list = []
+        value_list = []
+        for sample_score in scores:
+            assert sample_score.sample_metadata is not None
+            cluster_list.append(sample_score.sample_metadata[cluster])
```

Comment:
It looks like the cluster 'keys' are single level keys into metadata (so you can only cluster on keys at the root level). I think you probably considered supporting some sort of notation (e.g. `path.to.subkey`) and wanted to just briefly discuss / hear about whether it would be a good idea to support more here...